feathr_project/feathr/_databricks_submission.py

```diff
@@ -127,39 +127,52 @@ def submit_feathr_job(self, job_name: str, main_jar_path: str,  main_class_name:
+        submission_params['libraries'][0]['jar'] = self.upload_or_get_cloud_path(main_jar_path)
+        # see here for the submission parameter definition https://docs.microsoft.com/en-us/azure/databricks/dev-tools/api/2.0/jobs#--request-structure-6
+        if python_files:
+            # this is a pyspark job. definition here: https://docs.microsoft.com/en-us/azure/databricks/dev-tools/api/2.0/jobs#--sparkpythontask
+            param_and_file_dict = {"parameters": arguments, "python_file":self.upload_or_get_cloud_path(python_files[0])}
+            submission_params.setdefault('spark_python_task',param_and_file_dict)
+        else:
+            # this is a scala spark job
+            submission_params['spark_jar_task']['parameters'] = arguments
+            submission_params['spark_jar_task']['main_class_name'] = main_class_name
+
+
+        # For Job runs APIs, see https://docs.microsoft.com/en-us/azure/databricks/dev-tools/api/2.0/jobs#--runs-submit
+        result = requests.post(url=self.workspace_instance_url+'/api/2.0/jobs/runs/submit',
+                                headers=self.auth_headers, data=json.dumps(submission_params))
 
         try:
-            # For Job APIs, see https://docs.microsoft.com/en-us/azure/databricks/dev-tools/api/2.0/jobs
-            result = requests.post(url=self.workspace_instance_url+'/api/2.0/jobs/runs/submit',
-                              headers=self.auth_headers, data=json.dumps(submission_params))
+            # see if we can parse the returned result
             self.res_job_id = result.json()['run_id']
-            # For Job APIs, see https://docs.microsoft.com/en-us/azure/databricks/dev-tools/api/2.0/jobs
-            result = requests.get(url=self.workspace_instance_url+'/api/2.0/jobs/runs/get',
-                             headers=self.auth_headers, params={'run_id': str(self.res_job_id)})
-            logger.info('Feathr Job Submitted Sucessfully. View more details here: {}', result.json()[
-                        'run_page_url'])
         except:
-            traceback.print_exc()
+            logger.error("submitting to Databricks cluster failed. Message returned from Databricks: {}", result.json())
+            exit(1)
+
+        # For Job runs APIs, see https://docs.microsoft.com/en-us/azure/databricks/dev-tools/api/2.0/jobs#--runs-submit
+        result = requests.get(url=self.workspace_instance_url+'/api/2.0/jobs/runs/get',
+                                headers=self.auth_headers, params={'run_id': str(self.res_job_id)})
+        logger.info('Feathr Job Submitted Sucessfully. View more details here: {}', result.json()[
+            'run_page_url'])
+
         # return ID as the submission result
         return self.res_job_id
 
-    def wait_for_completion(self, timeout_seconds: Optional[int] = 500) -> bool:
+    def wait_for_completion(self, timeout_seconds: Optional[int] = 1200) -> bool:
```

Comment:
Why changing the default? Looks like it has no impact here, since no caller is using the default